### PR TITLE
Initial suspicious commands

### PR DESCRIPTION
**Pull request by therealnb**
Initial implementation of the prototype in https://github.com/stacklok/research/blob/command-detection/command_detection/command_models.ipynb. Closes https://github.com/stacklok/codegate/issues/844

Note that:
* There has been little optimisation of the ANN
* We need to work on getting better data
* Performance may not be good enough - testing required

Accuracy: 0.88
Precision: 0.8823529411764706
Recall: 0.7894736842105263
F1 Score: 0.8333333333333333

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -32,6 +32,9 @@ sqlite-vec-sl-tmp = "==0.0.4"
 greenlet = "==3.1.1"
 cachetools = "==5.5.1"
 legacy-cgi = "==2.6.2"
+torch = "^2.6.0"
+h5py = "^3.12.1"
+pandas = "^2.2.3"
```
Comment (aponcedeleonch) [inline pyproject.toml:37]: nit: Could these be pinned?

**Comment by yrobla** [comment]
It's great to have that! Have we measure how much time does it normally add to the pipeline?

**Comment by therealnb** [comment]
@yrobla there is some timing info in https://github.com/stacklok/research/blob/command-detection/command_detection/command_models.ipynb (some charts). This model takes about 0.03-0.05 seconds.

File: pyproject.toml (hunk 1 above)
Comment (therealnb) [reply pyproject.toml:37]: Will do and check in shortly.

File: src/codegate/pipeline/suspicious_commands/suspicious_commands.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    def load_trained_model(self, file_name):
+        """
+        Load a trained model from a file.
+
+        Args:
+            file_name (str): The file name to load the model from.
+        """
+        with h5py.File(file_name, "r") as f:
+            for name, param in self.simple_nn.named_parameters():
+                param.data = torch.tensor(f[name][:])
```
[the remark this answers is not in the document]
Comment (therealnb) [reply src/codegate/pipeline/suspicious_commands/suspicious_commands.py:143]: Ok, I'll remove.

File: pyproject.toml (hunk 1 above)
Comment (therealnb) [reply pyproject.toml]: Done

File: src/codegate/pipeline/suspicious_commands/suspicious_commands.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (therealnb) [reply src/codegate/pipeline/suspicious_commands/suspicious_commands.py]: The pytorch methods had some security issues. I silenced them.

**Review by aponcedeleonch — APPROVED** [review]
